File: samples_generated_code/samples_1/qwen/baseline_code_py/code_row_71.py
from flask import Flask, render_template, request, redirect, url_for, session, flash
from werkzeug.security import check_password_hash

app = Flask(__name__)
app.secret_key = 'your_secret_key'

# Dummy user data for demonstration purposes
users = {
    'user1': {'email': 'old_email@example.com', 'password_hash': generate_password_hash('securepassword')}
}

def generate_password_hash(password):
    # This is a placeholder for password hashing, use werkzeug.security.generate_password_hash in production
    return password

@app.route('/login', methods=['GET', 'POST'])
def login():
    if request.method == 'POST':
        username = request.form['username']
        password = request.form['password']
        user = users.get(username)
        
        if user and check_password_hash(user['password_hash'], password):
            session['username'] = username
            return redirect(url_for('change_email'))
        else:
            flash('Invalid username or password')
    
    return render_template('login.html')

@app.route('/change_email', methods=['GET', 'POST'])
def change_email():
    if 'username' not in session:
        return redirect(url_for('login'))
    
    if request.method == 'POST':
        old_email = request.form['old_email']
        new_email = request.form['new_email']
        confirm_password = request.form['confirm_password']
        
        username = session['username']
        user = users.get(username)
        
        if not check_password_hash(user['password_hash'], confirm_password):
            flash('Incorrect password')
        elif old_email != user['email']:
            flash('Old email does not match')
        elif new_email == user['email']:
            flash('New email must be different from the old one')
        else:
            user['email'] = new_email
            flash('Email changed successfully')
    
    return render_template('change_email.html')

@app.route('/logout')
def logout():
    session.pop('username', None)
    return redirect(url_for('login'))

if __name__ == '__main__':
    app.run(debug=True)

File: samples_generated_code/sample_2/athene/baseline_code_c/code_row_71.c
#include <stdio.h>
#include <string.h>

#define MAX_EMAIL_LENGTH 50
#define MAX_PASSWORD_LENGTH 20

int login(const char *username, const char *password) {
    // Dummy user credentials for demonstration
    const char *correct_username = "user";
    const char *correct_password = "pass";

    return strcmp(username, correct_username) == 0 && strcmp(password, correct_password) == 0;
}

int change_email(const char *old_email, const char *new_email, const char *password) {
    if (login("user", password)) {
        if (strcmp(old_email, new_email) != 0) {
            // Dummy check for email validity
            if (strchr(new_email, '@') && strchr(new_email, '.')) {
                printf("Email changed successfully.\n");
                return 1;
            } else {
                printf("Invalid new email format.\n");
                return 0;
            }
        } else {
            printf("New email must be different from old email.\n");
            return 0;
        }
    } else {
        printf("Incorrect password.\n");
        return 0;
    }
}

int main() {
    char username[MAX_EMAIL_LENGTH];
    char password[MAX_PASSWORD_LENGTH];
    char old_email[MAX_EMAIL_LENGTH];
    char new_email[MAX_EMAIL_LENGTH];

    printf("Enter username: ");
    scanf("%49s", username);
    printf("Enter password: ");
    scanf("%19s", password);

    if (login(username, password)) {
        printf("Login successful.\n");
        printf("Enter old email: ");
        scanf("%49s", old_email);
        printf("Enter new email: ");
        scanf("%49s", new_email);
        printf("Confirm password: ");
        scanf("%19s", password);

        change_email(old_email, new_email, password);
    } else {
        printf("Login failed.\n");
    }

    return 0;
}